fMRIVolume/scripts/MotionCorrection.sh

```diff
@@ -44,6 +59,28 @@ case "$MotionCorrectionType" in
     ;;
 esac
 
+
+# If registering across runs, perform nonlinear registration if requested.
+# (If using linear registration, don't need to do anything extra here, since
+# linear registration is handled implicitly via the motion correction).
+
+if [ "${fMRIReferenceReg}" == "nonlinear" ] ; then
+  verbose_echo " ... computing nonlinear transform to reference"
+  verbose_echo "     ... generating bold average"
+  # Generating a mean image to increase SNR to noise ratio when registering to scout.
+  ${FSLDIR}/bin/fslmaths ${WorkingDirectory}/${OutputfMRIBasename} -Tmean ${WorkingDirectory}/${OutputfMRIBasename}_avg
+
+  verbose_echo "     ... running fnirt: fnirt --in=${WorkingDirectory}/${OutputfMRIBasename}_avg --ref=${Scout} --refmask=${Scout}_mask --iout=${WorkingDirectory}/${OutputfMRIBasename}_avg_nonlin --cout=${WorkingDirectory}/mc2ref_warp"
```

Comment:
Does `${Scout}_mask` actually exist?  I'm not seeing where it would have been created...
And can we call the warp `postmc2fmriref_warp` just to make it a little more descriptive?

Reply:
Well, `${Scout}_mask` does exist, but it is only generated in `IntensityNormalization.sh`. We did not use `--refmask` before. I checked, if existed in the final file structure and put it in because of your previous comment why we are not using `-refmat`. It worked fine before w/o it. Is there a strong reason we should?

I did change the warp name to the suggested one.